dspace-api/src/main/java/org/dspace/content/RelationshipMetadataServiceImpl.java

```diff
@@ -0,0 +1,285 @@
+                                    virtualBean));
+                }
+            } else {
+                resultingMetadataValueList.addAll(
+                        findRelationshipMetadataValueFromBean(context, item, otherItem, relationship, place, key,
+                                virtualBean));
+            }
+        }
+        return resultingMetadataValueList;
+    }
+
+    /**
+     * This method will retrieve a list of RelationshipMetadataValue objects based on the config passed along in the
+     * hashmaps parameter. The beans will be used to retrieve the values for the RelationshipMetadataValue objects
+     * and the keys of the hashmap will be used to construct the RelationshipMetadataValue object.
+     *
+     * @param context               The context
+     * @param item                  The item whose virtual metadata is requested
+     * @param otherItem             The related item whose actual metadata is requested
+     * @param relationship          The relationship whose virtual metadata is requested
+     * @param place                 The place to use in the virtual metadata
+     * @param key                   The key corresponding to the VirtualMetadataConfiguration
+     * @param virtualBean           The VirtualMetadataConfiguration object which will generate the
+     *                              virtual metadata. This configuration is applicable for a relationship
+     *                              between both items
+     * @return                      The list of virtual metadata values
+     */
+    private List<RelationshipMetadataValue> findRelationshipMetadataValueFromBean(
+            Context context, Item item, Item otherItem, Relationship relationship, int place,
+            String key, VirtualMetadataConfiguration virtualBean) throws SQLException {
+        List<RelationshipMetadataValue> resultingMetadataValueList = new LinkedList<>();
+        for (String value : virtualBean.getValues(context, otherItem)) {
+            RelationshipMetadataValue relationshipMetadataValue = constructRelationshipMetadataValue(context, item,
+                                                                                                     relationship
+                                                                                                         .getID(),
+                                                                                                     place,
+                                                                                                     key, virtualBean,
+                                                                                                     value);
+            if (relationshipMetadataValue != null) {
+                resultingMetadataValueList.add(relationshipMetadataValue);
+            }
+        }
+        return resultingMetadataValueList;
+    }
+
+    private RelationshipMetadataValue constructRelationshipMetadataValue(Context context, Item item,
```

Comment:
Could you please add javadocs?

Reply:
This is a private methods, but some comments have been included